File: commands/structure.js
const chalk = require('chalk');
const { cloneRepo } = require('../lib/git');
const { traverseDir, saveOutput } = require('../lib/utils');

async function structure(repo, options) {
  try {
    console.log(chalk.blue(`\n🌳 Getting structure of ${repo}...\n`));

    // Clone repository
    const repoPath = await cloneRepo(repo, options.cacheDir, options.proxy);

    // Get structure
    const depth = parseInt(options.depth) || 3;
    const tree = traverseDir(repoPath, depth);

    // Display tree
    displayTree(tree);

    // Save if requested
    if (options.output) {
      saveOutput(tree, options.output);
      console.log(chalk.green(`\n✅ Structure saved to ${options.output}`));
    }

  } catch (error) {
    console.error(chalk.red(`\n❌ Error: ${error.message}`));
    process.exit(1);
  }
}

function displayTree(items, prefix = '') {
  for (let i = 0; i < items.length; i++) {
    const item = items[i];
    const isLastItem = i === items.length - 1;
    const connector = isLastItem ? '└─ ' : '├─ ';
    const icon = item.type === 'directory' ? '📁' : '📄';

    console.log(`${prefix}${connector}${icon} ${item.name}`);

    if (item.children && item.children.length > 0) {
      const newPrefix = prefix + (isLastItem ? '   ' : '│  ');
      displayTree(item.children, newPrefix);
    }
  }
}

module.exports = { structure };

File: cmd/structure.go
package cmd

import (
	"fmt"

	"github.com/spf13/cobra"
	"github.com/syxc/gh-repo-cli/internal/git"
	"github.com/syxc/gh-repo-cli/internal/utils"
)

var (
	structureDepth  int
	structureOutput string
)

// structureCmd represents the structure command
var structureCmd = &cobra.Command{
	Use:   "structure <repo>",
	Short: "Get repository directory structure",
	Long:  `Display the directory structure of a GitHub repository.`,
	Args:  cobra.ExactArgs(1),
	RunE: func(cmd *cobra.Command, args []string) error {
		return runStructure(args[0])
	},
}

func init() {
	rootCmd.AddCommand(structureCmd)
	structureCmd.Flags().IntVarP(&structureDepth, "depth", "d", 3, "Maximum depth")
	structureCmd.Flags().StringVarP(&structureOutput, "output", "o", "", "Save to file")
}

func runStructure(repo string) error {
	cfg := getConfig()

	fmt.Println(utils.Blue(fmt.Sprintf("\n🌳 Getting structure of %s...\n", repo)))

	// Clone repository
	repoPath, err := git.CloneRepo(repo, cfg.CacheDir, cfg.Proxy)
	if err != nil {
		utils.PrintError("%v", err)
		return err
	}

	// Get structure
	tree, err := utils.TraverseDir(repoPath, structureDepth, 0)
	if err != nil {
		utils.PrintError("%v", err)
		return err
	}

	// Display tree
	utils.DisplayTree(tree, "")

	// Save if requested
	if structureOutput != "" {
		if err := utils.SaveOutput(tree, structureOutput); err != nil {
			utils.PrintError("%v", err)
			return err
		}
		fmt.Println(utils.Green(fmt.Sprintf("\n✅ Structure saved to %s", structureOutput)))
	}

	return nil
}
